### ISSUE TITLE
imfile documentation incorrect WRT statefile naming

### ISSUE THREAD

**Issue by SirWumpus** [rsyslog/rsyslog#4324]
Current `imfile` documentation (https://rsyslog.readthedocs.io/en/latest/configuration/modules/imfile.html)  states this:

> As a concrete example, consider file /var/log/applog is being monitored. The corresponding state file will be named imfile-state:-var-log-applog.

Which is how I understood and knew it to be.   Example I'm currently working with:
```
imfile-state:-home-achowe-solo-var-tmp-events.txt
```
Following a recent set of updates and during testing I notice the state file appears to have been renamed into some sort of hash, which I assume is based on the file path.
```
imfile-state:140979:d499ed2e7ff38101
```
I want to rename the old file to the new, but cannot find documentation as to how to rename that file (how the hash was generated).  I cannot throw away the current file since its a specialised event  log and I cannot accept duplicates being reprocessed due to a reset.

### Workaround

I've read of the deprecated option `stateFile`, which I'll use to resolve this for the time being.

**Comment by rgerhards** [comment]
readthedocs is totally out of sync - don't use it (we tried to take the project over, but it never happened). Please read the doc at

https://www.rsyslog.com/doc/master/configuration/modules/imfile.html?highlight=imfile#state-files

It has the correct info and the answer to your question.

TL;DR: you do not need to rename the file manually, rsyslog detects this on startup and does do that automatically for you.

**Comment by rgerhards** [comment]
Just for the record: this is the readthedoc conversation: https://github.com/readthedocs/readthedocs.org/issues/6645
